java/src/main/java/com/company/automation/fundamentals/optional_task/optional_task_01/OptionalTask01.java

```diff
@@ -0,0 +1,414 @@
+                differentNumbers.append(number.charAt(i));
+            }
+        }
+
+        amountOfDifferentDigits = differentNumbers.toString().length();
+
+        return amountOfDifferentDigits;
+    }
+
+
+    private static int getQuantityOfIntegersWithOnlyEvenDigits(String[] numbers) {
+        int quantityOfNumbersWithOnlyEvenDigits = 0;
+
+        for (String val : numbers) {
+
+            if (hasOnlyEvenDigits(val)) {
+                quantityOfNumbersWithOnlyEvenDigits++;
+            }
+        }
+        return quantityOfNumbersWithOnlyEvenDigits;
+    }
+
+
+    private static boolean hasOnlyEvenDigits(String number) {
+        Pattern pattern = Pattern.compile("[13579]");
+        Matcher matcher = pattern.matcher(number);
+        return !matcher.find();
+    }
+
+
+    private static int getQuantityOfIntegersWithEqualAmountOfEvenAndUnevenDigits(String[] numbers) {
+        int quantityOfNumbersWithEqualAmountOfEvenAndUnevenDigits = 0;
+
+        for (String val : numbers) {
+
+            if (hasEqualAmountOfEvenAndUnevenDigits(val)) {
+                quantityOfNumbersWithEqualAmountOfEvenAndUnevenDigits++;
+            }
+        }
+
+        return quantityOfNumbersWithEqualAmountOfEvenAndUnevenDigits;
+    }
+
+
+    private static boolean hasEqualAmountOfEvenAndUnevenDigits(String number) {
+        String evenDigits;
+        String unevenDigits;
+        Matcher matcher;
+        StringBuilder digits = new StringBuilder();
+
+        Pattern even = Pattern.compile("[02468]");
+        Pattern uneven = Pattern.compile("[13579]");
+        matcher = even.matcher(number);
+
+        while (matcher.find()) {
+            digits.append(matcher.group());
+        }
+
+        evenDigits = digits.toString();
+
+        digits.delete(0, digits.length());
+        matcher = uneven.matcher(number);
+
+        while (matcher.find()) {
+            digits.append(matcher.group());
+        }
+
+        unevenDigits = digits.toString();
+
+        return evenDigits.length() == unevenDigits.length();
+    }
+
+
+    private static int getIntegerWithRisingDigitsSequence(String[] numbers) {
+        int integerWithRisingDigitsSequence = 0;
+
+        for (String val : numbers) {
+
+            if (hasRisingDigitsSequence(val)) {
+                integerWithRisingDigitsSequence = Integer.parseInt(val);
+                break;
+            }
+        }
+
+        return integerWithRisingDigitsSequence;
+    }
+
+
+    private static boolean hasRisingDigitsSequence(String number) {
+        boolean isRisingDigitsSequence = false;
+        int[] digitsOfTheNumber = getDigitsOfTheNumber(number);
+
+        for (int i = 0; i < digitsOfTheNumber.length - 1; i++) {
+            if (digitsOfTheNumber[i + 1] > digitsOfTheNumber[i]) {
+                isRisingDigitsSequence = true;
+            } else {
+                isRisingDigitsSequence = false;
+                break;
+            }
+        }
+
+        return isRisingDigitsSequence;
```

Comment:
private static boolean hasRisingDigitsSequence(String number) {
        int[] digitsOfTheNumber = getDigitsOfTheNumber(number);

        for (int i = 0; i < digitsOfTheNumber.length - 1; i++) {
            if (digitsOfTheNumber[i + 1] < digitsOfTheNumber[i]) {
                return false;
            } 
        }
        return true;
}